### ISSUE TITLE
[Research] Import/cloning tool for OpenStax books

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/ideas#440]
As a Pressbooks user, I want to be able to use Pressbooks to adapt and redistribute other high quality openly licensed content. Some of the best open textbooks available are those published by OpenStax: https://openstax.org/subjects

In the past, I believe OpenStax published these books using their CNX platform, and made zip files of books available. BCcampus wrote a now deprecated plugin that imported these books from the zip files and had an open issue to rewrite the plugin to make use of the CNX API: https://github.com/BCcampus/pressbooks-openstax-import/issues/71 + https://github.com/openstax/cnx-archive/blob/master/docs/content_api_doc.md

I don't know what platform is used to publish OpenStax titles currently, nor whether it has an API available. We should research this and produce a brief report on the feasibility of an API-based import/cloning routine from OpenStax to Pressbooks.

**Comment by SteelWagstaff** [comment]
Anthony Palmiotto from OpenStax wrote me on 2022.12.08: "Right now, we don't have an API but it might be in scope to develop one; it would not be very soon, however. ... [t]he materials are also available on repositories that you could also access. Should I give you that so you could evaluate it and see if it's a better option?"

I said yes and asked if we could meet to discuss future API plans.

**Comment by SteelWagstaff** [comment]
Anthony replied:
> We're not likely to develop an API on our own, and it might be best to talk to some folks who are directly engaged in the engineering tools for our outputs. I'm asking them now and will do an introduction if it's best. … [f]rom my understanding, epubs can be imported into Pressbooks.  Does that process work well for all types of content and textbook elements?

I replied:
> Happy to talk with your colleagues engaged in the engineering tools for your outputs. Our EPUB exporter is pretty good — not sure how well it would handle a very large book with lots of complex internal elements but happy to do some experiments if you have a few sample EPUB exports we could use?

**Comment by SteelWagstaff** [comment]
Still relevant. @cmurtagh and @jcurtis-pressbooks might be something worth exploring in 2025?

**Comment by jcurtis-pressbooks** [comment]
Yes, I have heard this request from multiple people. Every time OpenStax publishes new titles or significant updates to existing titles, this is going to come up. It's not surprising to see requests for LibreTexts book imports. 

I'm curious to understand the relative work required to support an easy OpenStax import, vs. an easy LibreTexts import, vs. an easy Google Doc import. Would these be comparable levels of effort? Are there any economies of scale in taking on multiple of these (seemingly related) features? What makes them harder vs. easier to achieve?

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 3y 3m ago with no updates in the last 1y 2m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542
